coalib/misc/Shell.py

```diff
@@ -102,7 +140,7 @@ def run_shell_command(command, stdin=None, **kwargs):
     :return:        A tuple with ``(stdoutstring, stderrstring)``.
     """
     with run_interactive_shell_command(command, **kwargs) as p:
-        ret = p.communicate(stdin)
+        ret = ShellCommandResult(p, stdin)
```

Comment:
I'm more for separating functionality, so let `ShellCommandResult` be a simple data-holder and do
```
for ...
    ret = p.communicate(stdin)
return ShellCommandResult(p.returncode, *ret)
```